core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -449,3 +456,81 @@ class DynamicLogConfig(logManager: LogManager) extends Reconfigurable with Loggi
     }
   }
 }
+
+class DynamicMetricsReporters(brokerId: Int, dynamicConfig: DynamicBrokerConfig) extends Reconfigurable {
+
+  private val propsOverride = Map[String, AnyRef](KafkaConfig.BrokerIdProp -> brokerId.toString)
+  private val currentReporters = mutable.Map[String, MetricsReporter]()
+  private var metrics: Metrics = _
+  initialize()
+
+  private def initialize() {
+    dynamicConfig.addReconfigurable(this)
+    val reporters = dynamicConfig.currentKafkaConfig.getList(KafkaConfig.MetricReporterClassesProp)
+    createReporters(reporters, Collections.emptyMap[String, Object], None)
+  }
+
+  private[server] def initMetrics(metrics: Metrics): Unit = {
+    this.metrics = metrics
+  }
+
+  private[server] def currentMetricsReporters: List[MetricsReporter] = currentReporters.values.toList
+
+  override def configure(configs: util.Map[String, _]): Unit = {}
+
+  override def reconfigurableConfigs(): util.Set[String] = {
+    val configs = new util.HashSet[String]()
+    configs.add(KafkaConfig.MetricReporterClassesProp)
+    currentReporters.values.foreach {
+      case reporter: Reconfigurable => configs.addAll(reporter.reconfigurableConfigs)
+      case _ =>
+    }
+    configs
+  }
+
+  override def validateReconfiguration(configs: util.Map[String, _]): Boolean = {
+    val updatedMetricsReporters = configs.get(KafkaConfig.MetricReporterClassesProp).asInstanceOf[util.List[String]].asScala
+
+    // Ensure all the reporter classes can be loaded and have a default constructor
+    updatedMetricsReporters.foreach { className =>
+      val clazz = Utils.loadClass(className, classOf[MetricsReporter])
+      clazz.getConstructor()
+    }
+
+    // Validate the new configuration using every reconfigurable reporter instance that is not being deleted
+    currentReporters.values.forall { reporter =>
```

Comment:
nit: maybe a little more natural to match on the type like above in `reconfigurableConfigs` so we don't need the instance check and cast.